File: Python/add_two_polynomials.py
def add_polynomials(p1, p2):
    length = max(len(p1), len(p2))
    result = [0] * length

    for i in range(len(p1)):
        result[i] += p1[i]

    for i in range(len(p2)):
        result[i] += p2[i]

    return result

# Example
poly1 = [3, 4, 5]  # 3 + 4x + 5x^2
poly2 = [1, 2]     # 1 + 2x

print("Sum:", add_polynomials(poly1, poly2))  # Output: [4, 6, 5]

File: C/add_two_polynomials.c
#include <stdio.h>

#define MAX 100

void addPolynomials(int a[], int b[], int m, int n) {
    int size = m > n ? m : n;
    int result[MAX];

    for (int i = 0; i < size; i++) {
        int coeffA = i < m ? a[i] : 0;
        int coeffB = i < n ? b[i] : 0;
        result[i] = coeffA + coeffB;
    }

    printf("Sum of polynomials:\n");
    for (int i = size - 1; i >= 0; i--) {
        if (result[i] != 0)
            printf("%dx^%d ", result[i], i);
    }
}

int main() {
    int m, n;
    printf("Enter degree+1 of first polynomial: ");
    scanf("%d", &m);
    int a[MAX];
    printf("Enter coefficients:\n");
    for (int i = 0; i < m; i++) scanf("%d", &a[i]);

    printf("Enter degree+1 of second polynomial: ");
    scanf("%d", &n);
    int b[MAX];
    printf("Enter coefficients:\n");
    for (int i = 0; i < n; i++) scanf("%d", &b[i]);

    addPolynomials(a, b, m, n);
    return 0;
}
